src/AppInstallerRepositoryCore/Microsoft/PredefinedWriteableSourceFactory.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+#include "Microsoft/ARPHelper.h"
+#include "Microsoft/PredefinedWriteableSourceFactory.h"
+#include "Microsoft/SQLiteIndex.h"
+#include "Microsoft/SQLiteIndexSource.h"
+#include <winget/ManifestInstaller.h>
+
+#include <winget/Registry.h>
+#include <AppInstallerArchitecture.h>
+
+using namespace std::string_literals;
+using namespace std::string_view_literals;
+
+namespace AppInstaller::Repository::Microsoft
+{
+    std::shared_ptr<ISource> PredefinedWriteableSourceFactoryImpl::g_sharedSource = nullptr;
+    std::once_flag PredefinedWriteableSourceFactoryImpl::g_InstallingSourceOnceFlag;
+
+    std::shared_ptr<ISource> PredefinedWriteableSourceFactoryImpl::Create(const SourceDetails& details, IProgressCallback& progress)
+    {
+        UNREFERENCED_PARAMETER(progress);
+
+        THROW_HR_IF(E_INVALIDARG, details.Type != PredefinedWriteableSourceFactory::Type());
+
+        std::string lockName = "WriteableSource_";
+
+        // Installing is the only type right now so just return the same source to all callers.
+        std::call_once(g_InstallingSourceOnceFlag,

```

Comment:
Good catch thanks, the higher layer stuff was added later and I didnt notice I could remove this.

Reply:
Actually, turns out this is needed. Everyone has to use the same instance because the index isn't shared across instances so people querying one instance wouldn't see writes to another instance if they had separate instances.